processor/src/main/java/com/ihren/processor/validation/contains/in/ContainsInStringValidator.java

```diff
@@ -0,0 +1,31 @@
+package com.ihren.processor.validation.contains.in;
+
+import com.ihren.processor.constant.Currency;
+import jakarta.validation.ConstraintValidator;
+import jakarta.validation.ConstraintValidatorContext;
+import java.util.Set;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+public class ContainsInStringValidator implements ConstraintValidator<ContainsIn, String> {
+
+    private final Set<String> allowedValues = Stream.of(Currency.values())
```

Comment:
There is no logic in on field initialization level.
It should be done config method